src/Core/Configuration/AvifExtensionChecker.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Configuration;
+
+use PrestaShop\PrestaShop\Core\Image\Exception\AvifUnavailableException;
+
+/**
+ * Class AvifExtensionChecker provides object-oriented way to check if AVIF extension is installed and available.
+ */
+final class AvifExtensionChecker
+{
+    private $psAdditionalImageAvif;
+
+    public function __construct($psAdditionalImageAvif)
+    {
+        $this->psAdditionalImageAvif = $psAdditionalImageAvif;
+    }
+
+    public function isAvailable()
+    {
+        $isAvailable = extension_loaded('gd') && /* @phpstan-ignore-line */
+            version_compare(PHP_VERSION, '8.1') >= 0 &&
+            $this->psAdditionalImageAvif &&
+            function_exists('imageavif') &&
+            is_callable('imageavif'); /* @phpstan-ignore-line */
+
+        // set_error_handler() is used to catch the warning generated by the imageavif() function
+        // @see https://stackoverflow.com/questions/71739530/php-8-1-imageavif-avif-image-support-has-been-disabled
+        if (function_exists('imageavif')) {
+            set_error_handler(
+                function ($severity, $message) { /* @phpstan-ignore-line */
+                    if (str_contains($message, 'imageavif(): AVIF image support has been disabled')) {
+                        throw new AvifUnavailableException($message);
+                    }
+                },
+                E_WARNING
+            );
+
+            try {
```

Comment:
So, to check if `avif` is there and it's working, there's a fake .avif file handled on every request. I'm not sure about it. I think it's overkill 👍

Reply:
I would just do a basic check in configuration and tell merchant to try it. He can switch it back if it does not work.

Reply:
You need to do this because of https://stackoverflow.com/questions/71739530/php-8-1-imageavif-avif-image-support-has-been-disabled

Reply:
the problem with avif is that you need to compile GD with avif support, it's not just a php extension to be added

So basically, most people are not ready for avif, and without proper check, it will fail for most people.

@Hlavtox we talked about this possibility with @MatShir but I don't think we like it in term of user experience ?

Reply:
This is a settings for a skilled person which are used about once per year, I don't think its worth it to invest time there. ;-)

Reply:
yes but... then the unskilled person will have the possibility to choose avif and won't understand why his BO is crashing when generating images